benchmark/Streamly/Benchmark/Prelude/Serial/Nested.hs

```diff
@@ -380,6 +381,156 @@ o_1_space_concat value = sqrtVal `seq`
 
     sqrtVal = round $ sqrt (fromIntegral value :: Double)
 
+-------------------------------------------------------------------------------
+-- Joining streams
+-------------------------------------------------------------------------------
+toKvMap :: Int -> (Int, Int)
+toKvMap p = (p, p)
+
+mkStreamLen :: (S.IsStream t, S.MonadAsync m) => Int -> t m Int
+mkStreamLen count = sourceUnfoldrM count 0
+
+{-# INLINE joinInner #-}
+joinInner :: (MonadIO m, S.MonadAsync m) => Int -> SerialT m Int -> m ()
+joinInner val strm =
+    if val > 0
```

Comment:
Do not use a if condition here. Just generate a stream of the `val` size`. See `concatMap` benchmarks which do the same.